src/controller/controller.ts

```diff
@@ -164,6 +166,7 @@ class Controller extends events.EventEmitter<ControllerEventMap> {
         // Register adapter events
         this.adapter.on('deviceJoined', this.onDeviceJoined.bind(this));
         this.adapter.on('zclPayload', this.onZclPayload.bind(this));
+        this.adapter.on('zdoResponse', this.onZdoResponse.bind(this));
         this.adapter.on('disconnected', this.onAdapterDisconnected.bind(this));
         this.adapter.on('deviceAnnounce', this.onDeviceAnnounce.bind(this));
```

Comment:
I think these can be removed now?

Reply:
Will do upstream removals in other PR, but yea, with a bit of clever work (zigate...), should be able to remove `deviceAnnounce` and `networkAddress` completely in favor of `zdoResponse` that triggers the same logic at the end.